src/DependenciesPackager/CrossgenContext.cs

```diff
@@ -179,13 +179,14 @@ private bool CrossgenAssembly(string publishFolder, PackageEntry package, Librar
             return exitCode == 0;
         }
 
-        private DirectoryInfo CreateSubDirectory(string cacheBasePath, PackageEntry package, LibraryAsset asset)
+        private DirectoryInfo CreateSubDirectoryInLowerCase(string cacheBasePath, PackageEntry package, LibraryAsset asset)
         {
             var subDirectoryPath = Path.Combine(
                 cacheBasePath,
                 package.Library.Identity.Name,
                 package.Library.Identity.Version.ToNormalizedString(),
-                Path.GetDirectoryName(asset.RelativePath));
```

Comment:
You should be using NuGet APIs to build these paths:
https://github.com/NuGet/NuGet.Client/blob/4cccb13833ad29d6a0bcff055460d964f1b49cfe/src/NuGet.Core/NuGet.Packaging/VersionFolderPathResolver.cs

Available in the NuGet.Packaging package:
https://dotnet.myget.org/gallery/nuget-build